frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/ApplicationMaster.java

```diff
@@ -605,15 +622,22 @@ private ContainerLaunchContext setupContainerLaunchContext(TaskStatus taskStatus
     return launchContext;
   }
 
+  // This function is to convert task's containerPorts from List<Range> format to string format
+  // The string format is "httpPort:80,81,82;sshPort:1021,1022,1023;"
+  private String setupPortsEnvironment(TaskStatus taskStatus) {
+    Map<String, Ports> portDefinitions = requestManager.getTaskResources().get(taskStatus.getTaskRoleName()).getPortDefinitions();
+    List<ValueRange> portRanges = taskStatus.getContainerPorts();
+    return ValueRangeUtils.toEnviromentString(portRanges, portDefinitions);
+  }
```

Comment:
We can directly call the ValueRangeUtils and remove the function here,

Reply:
ValueRangeUtil cannot handle the Map<String, Ports> portDefinitions. 

---
In reply to: [174683489](https://github.com/Microsoft/pai/pull/318#discussion_r174683489) [](ancestors = 174683489)

Reply:
Why it cannot handle

---
In reply to: [175338879](https://github.com/Microsoft/pai/pull/318#discussion_r175338879) [](ancestors = 175338879,174683489)